hack/common.sh

```diff
@@ -0,0 +1,386 @@
+     if [ "${ip_family}" = "ipv6" ]; then
+       set_sysctl net.ipv6.conf.all.forwarding 1
+       set_sysctl net.bridge.bridge-nf-call-arptables 0
+       set_sysctl net.bridge.bridge-nf-call-ip6tables 0
+       set_sysctl net.bridge.bridge-nf-call-iptables 0
+     fi
+}
+
+function install_calico {
+    ###########################################################################
+    # Description:                                                            #
+    # Copy binaries from the net to binaries directory                        #
+    #                                                                         #
+    # Arguments:                                                              #
+    #   arg1: binary directory, path to where ginko will be installed         #
+    ###########################################################################
+
+    local bin_directory="${1}"
+
+    ### Cache cni images to avoid rate-limiting
+    docker pull docker.io/calico/kube-controllers:v3.19.1
+    docker pull docker.io/calico/cni:v3.19.1
+    docker pull docker.io/calico/pod2daemon-flexvol:v3.19.1
+
+    if [ "$bin_directory" = "" ];
+    then
+        kind load docker-image docker.io/calico/cni:v3.19.1 --name ${CLUSTER_NAME}
+        kind load docker-image docker.io/calico/kube-controllers:v3.19.1 --name ${CLUSTER_NAME}
+        kind load docker-image docker.io/calico/pod2daemon-flexvol:v3.19.1 --name ${CLUSTER_NAME}
+
+        kubectl apply -f https://raw.githubusercontent.com/jayunit100/k8sprototypes/master/kind/calico.yaml
+        kubectl -n kube-system set env daemonset/calico-node FELIX_IGNORELOOSERPF=true
+        kubectl -n kube-system set env daemonset/calico-node FELIX_XDPENABLED=false
+    else
+        ${bin_directory}/kind load docker-image docker.io/calico/cni:v3.19.1 --name kpng-proxy
+        ${bin_directory}/kind load docker-image docker.io/calico/kube-controllers:v3.19.1 --name kpng-proxy
+        ${bin_directory}/kind load docker-image docker.io/calico/pod2daemon-flexvol:v3.19.1 --name kpng-proxy
+
+        ${bin_directory}/kubectl apply -f https://raw.githubusercontent.com/jayunit100/k8sprototypes/master/kind/calico.yaml
+        ${bin_directory}/kubectl -n kube-system set env daemonset/calico-node FELIX_IGNORELOOSERPF=true
+        ${bin_directory}/kubectl -n kube-system set env daemonset/calico-node FELIX_XDPENABLED=false
+    fi   
+}
+
+function compile_bpf {
+    ###########################################################################
+    # Description:                                                            #
+    # compile bpf elf files for ebpf backend                                  #
+    ###########################################################################
+
+    pushd "${SCRIPT_DIR}"/../backends/ebpf > /dev/null || exit
+    make bytecode
+    if_error_exit "Failed to compile EBPF Programs"
+    popd > /dev/null || exit
+
+
+    pass_message "Compiled BPF programs"
+}
+
+
+
```

Comment:
no need extra lines here

Reply:
> Another question:
> 
> `$ make tilt-setup i=ipv4 b=iptables`
> 
> If it's used iptables as backend, is it worth installing bpg2to ?
> 
> ```
> [ INFO ] 'bpf2go' not found, installing with 'go'
> go: downloading github.com/cilium/ebpf v0.9.2
> go: downloading golang.org/x/sys v0.0.0-20220829200755-d48e67d00261
> # internal/race
> compile: version "go1.17.6" does not match go tool version "go1.18.8"
> # internal/unsafeheader
> compile: version "go1.17.6" does not match go tool version "go1.18.8"
> # runtime/internal/sys
> compile: version "go1.17.6" does not match go tool version "go1.18.8"
> # internal/goexperiment
> compile: version "go1.17.6" does not match go tool version "go1.18.8"
> # unicode/utf8
> compile: version "go1.17.6" does not match go tool version "go1.18.8"
> # internal/itoa
> compile: version "go1.17.6" does not match go tool version "go1.18.8"
> # math/bits
> compile: version "go1.17.6" does not match go tool version "go1.18.8"
> # unicode
> compile: version "go1.17.6" does not match go tool version "go1.18.8"
> # encoding
> compile: version "go1.17.6" does not match go tool version "go1.18.8"
> # golang.org/x/sys/internal/unsafeheader
> compile: version "go1.17.6" does not match go tool version "go1.18.8"
> # internal/abi
> compile: version "go1.17.6" does not match go tool version "go1.18.8"
> # sync/atomic
> compile: version "go1.17.6" does not match go tool version "go1.18.8"
> # internal/cpu
> compile: version "go1.17.6" does not match go tool version "go1.18.8"
> # runtime/internal/atomic
> compile: version "go1.17.6" does not match go tool version "go1.18.8"
> [ FAILED ] cannot install bpf2go
> ```

@dougsland Currently all the setup are done when running `make tilt-setup`.  We only build images and redeploy once the `Tilt` server is started.   If we install `bpf2go` only when users try to use `ebpf` backend , We may have to increase `Tilt` timeout as well. If the `Tilt` was not able to trigger an re-deployment within the `timeout`, it'll fail.

I think the expectation (as per https://github.com/kubernetes-sigs/kpng/issues/383#issue-1410480330)  is that user will start with one backend and later they'll change.